app/products/calls/components/current_call/index.ts

```diff
@@ -16,14 +16,12 @@ import type {GlobalState} from '@mm-redux/types/store';
 function mapStateToProps(state: GlobalState) {
     const currentCall = getCurrentCall(state);
     const currentUserId = getCurrentUserId(state);
-    const speakerId = currentCall && currentCall.speakers && currentCall.speakers[0];
-    const speaker = currentCall && ((speakerId && currentCall.participants[speakerId]) || Object.values(currentCall.participants)[0]);
     const currentParticipant = currentCall?.participants[currentUserId];
+
     return {
         theme: getTheme(state),
         call: currentCall,
-        speaker,
-        speakerUser: speaker ? state.entities.users.profiles[speaker.id] : null,
+        profiles: state.entities.users.profiles,
```

Comment:
Nothing to change here, hopefully this is not as scary as it looks but be aware that this could cause the component to re-render like crazy

Reply:
That's a good point which I didn't fully consider. I'll see if I can improve it a bit.

Reply:
I've embedded the profile in the call's participant struct. Hopefully this should avoid the potential extra re-renders. Let me know if it makes sense.